File: java/12909.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<Character>();
        
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                stack.push(s.charAt(i));
            } else if (s.charAt(i) == ')') {
                if (stack.isEmpty()) {
                    return false;
                } else {
                    stack.pop();
                }
            }
        }
        
        if (stack.size() > 0) {
            return false;
        }

        return true;
    }
}

File: python/12909.py
from collections import deque

def solution(s):
    answer = True
    stack = deque()
    
    for i in s:
        if not stack or i == '(':
            stack.append(i)
        else: 
            if stack[-1] == '(':
                stack.pop()
            else:
                stack.append(i)
    
    return len(stack) == 0